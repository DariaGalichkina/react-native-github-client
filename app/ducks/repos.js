
import apiProvider from '../lib/apiProvider.js';

const SET_DATA = 'repos/SET_DATA';

export default function reducer(state = null, action) {

  switch (action.type) {

    case SET_DATA:
      return action.reposData;

    default:
      return state;

  }

}

export function getRepos() {

  return async (dispatch) => {
    const reposData = await apiProvider.getRepos();

    dispatch(setReposData(reposData));
  };
}

export function setReposData(reposData) {
  return { type: SET_DATA, reposData };
}
