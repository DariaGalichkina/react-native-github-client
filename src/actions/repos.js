import GitHubAPI from '../storage/GitHubAPI';

export function reposIsLoading(bool) {
    return {
        type: 'REPOS_IS_LOADING',
        isLoading: bool
    };
}

export function reposHasErrored(bool) {
    return {
        type: 'REPOS_HAS_ERRORED',
        hasErrored: bool
    }
}

export function reposFetchDataSuccess(repos) {
    return {
        type: 'REPOS_FETCH_DATA_SUCCESS',
        repos
    }
}

export function reposFetchData(url) {
    return async (dispatch) => {
        dispatch(reposIsLoading(true));

        var repos = await GitHubAPI.doApiCall('/user/repos');
        console.log(repos);
        if(repos) {
            dispatch(reposFetchDataSuccess(repos));
            dispatch(reposIsLoading(false));
        } else {
            dispatch(reposHasErrored(true));
            dispatch(reposIsLoading(false));
        }

    }
}