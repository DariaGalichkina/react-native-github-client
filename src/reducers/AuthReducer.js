import {
    LOGIN_CHANGED,
    PASSWORD_CHANGED,
    LOGIN_USER_SUCCESS,
    LOGIN_USER_FAIL,
    LOGIN_USER
} from '../actions/types';

const INITIAL_STATE = {
    login: '',
    password: '',
    user: null,
    error: '',
    loading: false
};

export default (state = INITIAL_STATE, action) => {
    switch (action.type) {
        case LOGIN_CHANGED:
            return { ...state, login: action.payload };
        case PASSWORD_CHANGED:
            return { ...state, password: action.payload };
        case LOGIN_USER:
            return { ...state, loading: true, error: '' };
        case LOGIN_USER_SUCCESS:
            return { ...state, user: action.payload, loading: false };
        case LOGIN_USER_FAIL:
            return { ...state, error: 'Login and/or password are incorrect', password: '', loading: false };
        default:
            return state;
    }
};
