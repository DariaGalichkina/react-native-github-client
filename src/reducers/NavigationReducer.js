import { NavigationActions } from 'react-navigation';
import {
    LOGIN_USER_SUCCESS,
    LOGIN_USER_FAIL,
} from '../actions/types';

import { AppNavigator } from '../containers/AppNavigator';

const firstAction = AppNavigator.router.getActionForPathAndParams('Login');
const loginNavState = AppNavigator.router.getStateForAction(firstAction);
//const secondAction = AppNavigator.router.getActionForPathAndParams('Repositories');

export default (state = loginNavState, action) => {
    let nextState;
    switch (action.type) {
        case LOGIN_USER_SUCCESS:
            nextState = AppNavigator.router.getStateForAction(
                NavigationActions.navigate({ routeName: 'Repositories'}),
                state
            );
            break;
        case LOGIN_USER_FAIL:
            nextState = AppNavigator.router.getStateForAction(
                NavigationActions.navigate({ routeName: 'Login' }),
                state
            );
            break;
        default:
            nextState = AppNavigator.router.getStateForAction(action, state);
            break;
    }

    return nextState || state;
}