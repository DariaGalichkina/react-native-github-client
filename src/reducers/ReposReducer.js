import {
    REPOSITORIES_FETCH,
    REPOSITORIES_FETCH_SUCCESS,
    REPOSITORIES_FETCH_FAIL
} from '../actions/types';

const INITIAL_STATE = {
    repositories: [],
    error: '',
    loading: false
};

export default (state = INITIAL_STATE, action) => {
    switch (action.type) {
        case REPOSITORIES_FETCH:
            return { ...state, loading: true, error: '' };
        case REPOSITORIES_FETCH_SUCCESS:
            return { ...state, repositories: action.payload, loading: false };
        case REPOSITORIES_FETCH_FAIL:
            return { ...state, error: 'Repositories are incorrect', password: '', loading: false };
        default:
            return state;
    }
};
