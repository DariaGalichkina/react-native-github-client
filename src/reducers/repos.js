export function reposHasErrored(state = false, action) {
    switch (action.type) {
        case 'REPOS_HAS_ERRORED':
            return action.hasErrored;

        default:
            return false;
    }
}

export function reposIsLoading(state = false, action) {
    switch (action.type) {
        case 'REPOS_IS_LOADING':
            return action.isLoading;

        default:
            return false;
    }
}

export function repos(state = [], action) {
    switch (action.type) {
        case 'REPOS_FETCH_DATA_SUCCESS':
            return action.repos;

        default:
            return state;
    }
}