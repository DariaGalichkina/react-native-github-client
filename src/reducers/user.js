export function userHasErrored(state = false, action) {
    switch (action.type) {
        case 'USER_HAS_ERRORED':
            return action.hasErrored;

        default:
            return false;
    }
}

export function userIsLoading(state = false, action) {
    switch (action.type) {
        case 'USER_IS_LOADING':
            return action.isLoading;

        default:
            return false;
    }
}

export function user(state = [], action) {
    switch (action.type) {
        case 'USER_LOGIN_SUCCESS':
            return action.auth;

        default:
            return state;
    }
}