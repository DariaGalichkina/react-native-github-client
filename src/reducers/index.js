import { combineReducers } from 'redux';
import AuthReducer from './AuthReducer';
import ReposReducer from './ReposReducer';
import NavigationReducer from './NavigationReducer';

const AppReducer = combineReducers({
    auth: AuthReducer,
    repos: ReposReducer,
    nav: NavigationReducer,
});

export default AppReducer;