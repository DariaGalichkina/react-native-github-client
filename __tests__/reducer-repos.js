import 'react-native';
import React from 'react';
import { repos, reposHasErrored, reposIsLoading } from '../src/reducers/repos';

it('returns correct state', () => {
    expect(repos(null, { type: 'REPOS_FETCH_DATA_SUCCESS', repos: [] })).toMatchSnapshot();
});

it('returns bool', () => {
    expect(reposHasErrored(null, { type: 'REPOS_HAS_ERRORED', hasErrored: true })).toMatchSnapshot();
});

it('returns bool', () => {
    expect(reposIsLoading(null, { type: 'REPOS_IS_LOADING', isLoading: true })).toMatchSnapshot();
});
