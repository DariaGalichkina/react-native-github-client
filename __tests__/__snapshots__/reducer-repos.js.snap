// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns bool 1`] = `true`;

exports[`returns bool 2`] = `true`;

exports[`returns correct state 1`] = `Array []`;
